lib/irc/Scheduler.js

```diff
@@ -0,0 +1,107 @@
+/*eslint no-invalid-this: 0*/
+"use strict"
+var promiseutil = require("../promiseutil");
+var Promise = require("bluebird");
+var logging = require("../logging");
+var log = logging.get("scheduler");
+
+function ScheduledQueue (server) {
```

Comment:
Overall this looks dangerously close to being the same as `Queue` in the util file, but with a set consume interval rather than consume-as-fast-as-possible. It would be better if you just added the extra config option (consume-at-intervals) to the `Queue` class rather than redo what the `Queue` is doing in the first place.